src/compiler/checker.ts

```diff
@@ -31650,7 +31665,7 @@ namespace ts {
             if (!symbol || !(symbol.flags & SymbolFlags.Function)) {
                 return false;
             }
-            return !!forEachEntry(getExportsOfSymbol(symbol), p => p.flags & SymbolFlags.Value && isPropertyAccessExpression(p.valueDeclaration));
+            return !!forEachEntry(getExportsOfSymbol(symbol), p => p.flags & SymbolFlags.Value && p.valueDeclaration && isPropertyAccessExpression(p.valueDeclaration));

```

Comment:
so an ambient class sets SymbolFlags.Value too? I assume that's why we have to check `p.valueDeclaration` now, right?

Reply:
Ambient classes do, in fact, count as values, since they do, in fact, resolve in an expression. A declaration being ambient just means it's missing a body - doesn't change its meaning in any other way, really.

Reply:
(The value without a value declaration thing shouldn't happen anymore - it's what caused me to do the "Assignment no longer implies Type or Value" in the other PR for `@enum` I merged last week. The reason I had to add this was because I had a namespace with no value declaration due to an assignment-caused namespace in a test, IIRC.)

Reply:
So it sounds like you can revert if it you want, or keep it for future safety. Either way is fine with me.

Reply:
I'd rather keep it - I always feel assuming `valueDeclaration` exists is sketchy anyway, and there's minimal overhead to checking it here.